### PR TITLE
feat: Project Module

### PR DESCRIPTION
**Pull request by cbolles**
# Description

Add in a project module

* Create new project
* Get a list of projects
* Get detailed project information
* Update project settings

## Checklist

- [x] This PR can be reviewed in under 30 minutes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have assigned reviewers to this PR.

### CODE REVIEW

File: packages/server/prisma/schema.prisma [hunk 1]
```diff
@@ -24,3 +24,16 @@ model User {
...
 }
+
+model Project {
+  id          String @default(uuid()) @id
+  name        String
+  description String?
+  logo        String?
+  muiTheme    Json?
+  createdAt   DateTime  @default(now())
+  updatedAt   DateTime  @default(now())
```
Comment (gregfrasco) [inline packages/server/prisma/schema.prisma]: Change to 
```suggestion
  updatedAt DateTime @updatedAt
```

File: packages/server/src/project/project.controller.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Controller, Post, Get, Body, Param, Put } from '@nestjs/common';
+import { Prisma, Project } from '@prisma/client';
+import { ProjectIdentifier, ConfigurableProjectSettings } from './dto/project.dto';
+import { ProjectService } from './project.service';
+
+@Controller('/project')
```
Comment (gregfrasco) [inline packages/server/src/project/project.controller.ts]: To be restful
```suggestion
@Controller('/projects')
```

File: packages/server/src/project/project.controller.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export class ProjectController {
+  constructor(private projectService: ProjectService) {}
+
+  @Post('/create')
+  createProject(@Body() newProject: Prisma.ProjectCreateInput): Promise<Project> {
+    return this.projectService.createProject(newProject);
+  }
+
+  @Get('/list')
```
Comment (gregfrasco) [inline packages/server/src/project/project.controller.ts]: `/projects` should just list projects 
```suggestion
  @Get()
```

File: packages/server/src/project/project.service.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  /**
+   * Create a new project. If the project ID is provided and a project with
+   * the same ID exists, will throw an error.
+   *
+   * @param newProject Information on the project to create
+   * @returns The newly created project
+   * @throws Will throw an error if the project ID is already in use
+   */
+  async createProject(newProject: Prisma.ProjectCreateInput): Promise<Project> {
+    if (await this.exists(newProject.id)) {
```
Comment (gregfrasco) [inline packages/server/src/project/project.service.ts]: When creating, id should be undefined since it is auto generated

**Review by gregfrasco — COMMENTED** [review]
Minor changes

**Review by gregfrasco — APPROVED** [review]

**Review by wenhwang97 — APPROVED** [review]

**Pull request merged**